tests/network/primary_pod_network.go

```diff
@@ -107,7 +107,10 @@ var _ = SIGDescribe("[Serial]Primary Pod Network", func() {
 				It("should report PodIP/s as its own on interface status", func() {
 					vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, vmi.Namespace)
 					Eventually(vmiIP).Should(WithTransform(removeCIDR, Equal(vmiPod.Status.PodIP)), "should contain VMI Status IP as Pod status ip")
-					Eventually(vmiIPs).Should(ContainElements(vmi.Status.Interfaces[0].IP, libnet.DefaultIPv6Address), "should contain at IPs the IP and the cloud-init configured ipv6")
+
+					// Check first fixed IPv6 address so we are sure that GA has quick in
+					Eventually(vmiIPs).Should(ContainElement(libnet.DefaultIPv6Address), "should contain IPv6 address set by cloud-init and reported by guest agent")
+					Eventually(vmiIPs).Should(ContainElement(vmi.Status.Interfaces[0].IP), "should contain IPv4 reported by guest agent")
```

Comment:
I thought that the problem was with this `vmi.Status` which may not be updated with the `vmiIPs` refresh. (I guess it depends when it is resolved, which may be compilation dependent)

Reply:
Eventually will just repeat what is inside it, but the Assert is read once at the beginning, so first I make sure that GA has run checking IPv6 so vmi Status is updated then I check the IPv4

Reply:
Yes, but that is a workaround to the real problem. Notice that there is no need to use Eventually now on the second assert as you already handled the race on the first.

I find this `vmiIPs` refreshing the `vmi` variable a side effect which is hard to understand when reading the test, so I like it less.
Someone may also see this code later on and try to refactor it to the original version here.

I'm in a conflict here.. on one hand just removing the 2nd `Eventually` may make this "ok", on the other hand if we look to the future, it may cause an author to repeat this problem again.

Reply:
Please consider that since `vmiIPs` refreshes the VMI variable, it will be prone for the randomly "missing status" problem, in case `Eventually` isn't used, and result in flakiness.

The bug of kubevirt from what i saw, is that even the IPs are reported in VMI status, they might disappear and come back after up to 2 minutes in the steady state.
It is caused due to a problem of kubevirt update VMI status logic, for the Guest agent reported part.

I would suggest to leave the `Eventually` or reuse the VMI being fetched.
Unless we do want to see this flakiness and then the bug will be officially covered by CI, which in turn can be a good idea.

Reply:
First, this PR targets to handle the race condition between the GA condition being connected and the update of the GA info into the VMI update. This is now resolved by the first line.
So any other issue here is not related to this change and will require probably some other treatment.
The issue of the IP disappearing is already handled in another PR, by using cloud-init and if there are other issues then other changes are needed.

The 2nd line will happen immediately after the first, which means it is unlikely to be a problem here of the VMI getting updated. But, as Eventually is not needed, also re-using the fetch is not needed, so it should be obvious that `vmi.Status.Interfaces[0].IPs` can be used directly.

> Simply add Consistently (for couple minutes, re-fetching and verifying IPs are there) after IPs found once in the VMI status, and you will see the problem.

We are not interested to increase the test time. After cloud-init network-data is used, you can try to recreate this and then we can consider a dedicated long test for this.

Reply:
> First, this PR targets to handle the race condition between the GA condition being connected and the update of the GA info into the VMI update. This is now resolved by the first line.
> So any other issue here is not related to this change and will require probably some other treatment.
> The issue of the IP disappearing is already handled in another PR, by using cloud-init and if there are other issues then other changes are needed.

Just to be synced please:

I am not talking about the problem of IP disappearing because we didn't use nmcli proper commands / networkData,
which i believe is what you mean (https://github.com/kubevirt/kubevirt/pull/4448)
I am talking about the random missing reports (aka hiccups).

Let's simplify it to what i saw (without even involving IPs):

* Take a healthy VM.
* Wait the 2nd interface to appear.
* Consistently re-fetch VMI status and check interface is there.

I saw the interface gone and comes back.
And when i had IPs i set manually using proper nmcli commands, i saw another scenario where the interface is there
but the IPs are gone and back (the VM had the IP fine all the time in this scenario, just the VMI status didn't).

> 
> The 2nd line will happen immediately after the first, which means it is unlikely to be a problem here of the VMI getting updated. But, as Eventually is not needed, also re-using the fetch is not needed, so it should be obvious that `vmi.Status.Interfaces[0].IPs` can be used directly.

I saw the problem couple times, where there was a re-fetch, as long as there isn't a re-fetch, it will not be flaky.

> 
> > Simply add Consistently (for couple minutes, re-fetching and verifying IPs are there) after IPs found once in the VMI status, and you will see the problem.
> 
> We are not interested to increase the test time. After cloud-init network-data is used, you can try to recreate this and then we can consider a dedicated long test for this.

Of course i meant just for us to see the scenario while developing.